### PR TITLE
Feature/mx 1583 make asset dependencies explicit

### PR DESCRIPTION
**Pull request by rababerladuseladim**
# PR Context

Make dependent extractors explicitly dependent on each other.

# Changes

- make dependent extractors explicitly depend on each other (gippeweb on confluence-vvt, biospecimen on synopse, odk on international-projects) 

# Fixed

- fix confluence_vvt transformation: use interne Vorgangsnummer as identifierInPrimarySource

### CODE REVIEW

File: mex/confluence_vvt/main.py [hunk 1]
```diff
@@ -77,23 +79,25 @@ def extracted_confluence_vvt_person_ids_by_query_string(
 
 
 @asset(group_name="confluence_vvt")
-def extracted_confluence_vvt_sources(
+def extracted_confluence_vvt_activities(
```
Comment (cutoffthetop) [inline mex/confluence_vvt/main.py:82]: oh, still some model v1 relics..

File: mex/confluence_vvt/transform.py [hunk 2]
```diff
@@ -7,13 +7,13 @@
 
 
 @watch
-def transform_confluence_vvt_sources_to_mex_sources(
+def transform_confluence_vvt_sources_to_mex_activities(
```
Comment (cutoffthetop) [inline mex/confluence_vvt/transform.py:10]:
```suggestion
def transform_confluence_vvt_sources_to_extracted_activities(
```

maybe even this, while we're at it

File: mex/confluence_vvt/transform.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def transform_confluence_vvt_sources_to_mex_sources(
             or involved_person_merged_ids
             or responsible_unit_merged_ids,
             documentation=source.documentation,
-            identifierInPrimarySource=source.identifier,
+            identifierInPrimarySource=source.identifier_in_primary_source,
```
Comment (cutoffthetop) [inline mex/confluence_vvt/transform.py:66]: looks like the "source" model is already transformed :(

i created MX-1631 to clean that up.

File: tests/biospecimen/test_transform.py [hunk 4]
```diff
@@ -1,34 +1,92 @@
...
     extracted_primary_sources: dict[str, ExtractedPrimarySource],
     biospecimen_resources: BiospecimenResource,
     mex_persons: list[ExtractedPerson],
     extracted_organization_rki: ExtractedOrganization,
+    extracted_synopse_activities: list[ExtractedActivity],
 ) -> None:
     unit_stable_target_ids = MagicMock()
     unit_stable_target_ids.get.side_effect = lambda _: Identifier.generate(seed=42)
 
+    synopse_merged_id = [  # noqa: RUF015
```
Comment (cutoffthetop) [inline tests/biospecimen/test_transform.py:77]: i mean, just a test, why not next()?

File: tests/confluence_vvt/test_transform.py [hunk 5]
```diff
@@ -93,14 +95,16 @@ def test_transform_confluence_vvt_source_items_to_mex_source_mocked(
...
+        transform_confluence_vvt_sources_to_mex_activities(
+            confluence_vvt_sources,
+            confluence_primary_source,
+            fake_authors,
+            unit_merged_ids_by_synonym,
+        )
     )
 
-    mex_source = next(mex_sources)
+    mex_activity = mex_activities[0]
```
Comment (cutoffthetop) [inline tests/confluence_vvt/test_transform.py:107]: next no good?

**Review by cutoffthetop — CHANGES_REQUESTED** [review]
very nice tidying up 🧹✨ just tiny remarks.

File: tests/biospecimen/test_transform.py (hunk 4 above)
Comment (rababerladuseladim) [reply tests/biospecimen/test_transform.py]: next does not work on list comprehensions, no idea why ruff suggests it here. Anyway, replaced the list comprehension with `filter` in 9e86f2a4867091c439a0839acc5e39346934f552

File: tests/confluence_vvt/test_transform.py (hunk 5 above)
Comment (rababerladuseladim) [reply tests/confluence_vvt/test_transform.py:107]: next does not work on list comprehensions, no idea why ruff suggests it here. Anyway, replaced the list comprehension with `filter` in 9e86f2a4867091c439a0839acc5e39346934f552

File: mex/confluence_vvt/transform.py (hunk 2 above)
Comment (rababerladuseladim) [reply mex/confluence_vvt/transform.py]: done in 01db3ac38a89886407e5d6c1f5228dae6d3b8377

**Review by cutoffthetop — APPROVED** [review]

**Pull request merged**